### DISCUSSION TITLE
Controlling source visibility in a sub scene

### DISCUSSION THREAD

**Discussion by phill-85** [Palakis/obs-websocket#1159] [Q&A]
I have 3 cameras overlaid on top of each other in a scene. Lets call it "camera scene".

I then use the "camera scene" in other scenes as an input. for example it might look something like this:

```
Scene 1
├ Masking Graphic
└ Camera Scene
     ├ Camera 1
     ├ Camera 2
     └ Camera 3
```

Is it possible, using websockets, to show and hide Camera 1/2/3 when the scene is in preview?

I'm using bitfocus companinon. If i show / hide the cameras in the "camera scene" then the live also updates.

**Comment by gbosky** [comment]
I have the same sort of configuration with my window captures. Did you ever find a way to toggle the sources for each scene in preview?

EDIT: I guess I needed to comment to find the solution, but at least I can pass on the help!

What I got to work was using a "get" to find the scene item id and then a "set" scene item enabled using that id. Here is my following code in python:

```
yy = cl.get_scene_item_id('Scene 1', 'Window 1 Border')
zz = cl.set_scene_item_enabled('Scene 1', yy.scene_item_id, True)
```

Note: yy return an integer 83 for my case. 

It finally clicked to me that the visibility was related to the scene, so I started looking at the scene commands for something. I wish there was a toggle function instead but at least its something that I can solve with a few more lines myself.

**Reply by Gamer1990** [reply]
in obs use Duplicated Scene than it may work not sure for sure as i use obs ws ahk based on 5ony work and have automated multiple commands into 1 Button {down} /{up} to trigger everything
